Include examples and benchmarks in packaged wheel

Navya1707 opened the pull request:
https://github.com/LMCache/LMCache/issues/2066
What this PR does / why we need it:

Package examples and benchmarks into the lmcache wheel by copying them under lmcache/ and adding them to MANIFEST.in + package_data so installed users get the resources.

Special notes for reviewers:

Build verified with NO_CUDA_EXT=1 python -m build --wheel; wheel includes examples/benchmarks.
No code logic changes; only packaging/resource inclusion.

Navya1707 committed:
Include examples and benchmarks in packaged whee

Signed-off-by: Navya Srivastava <navya.srivastava1707@gmail.com>

maobaolong reviewed (approved):
@Navya1707 Great job!

maobaolong commented:
mypy failed
```
Check SPDX headers.......................................................Passed
isort....................................................................Passed
ruff.....................................................................Passed
ruff-format..............................................................Passed
codespell................................................................Passed
clang-format.............................................................Passed
mypy.....................................................................Failed
- hook id: mypy
- exit code: 2

lmcache/benchmarks/ttft-estimator/ttft-estimator.py: error: Duplicate module named "ttft-estimator" (also at "benchmarks/ttft-estimator/ttft-estimator.py")
lmcache/benchmarks/ttft-estimator/ttft-estimator.py: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#mapping-file-paths-to-modules for more info
lmcache/benchmarks/ttft-estimator/ttft-estimator.py: note: Common resolutions include: a) using `--exclude` to avoid checking one of them, b) adding `__init__.py` somewhere, c) using `--explicit-package-bases` or adjusting MYPYPATH
Found 1 error in 1 file (errors prevented further checking)
```

sammshen commented:
not sure if we actually want this, will come back to this later

Navya1707 committed:
Include examples and benchmarks in packaged wheel

Signed-off-by: Navya Srivastava <navya.srivastava1707@gmail.com>

Navya1707 commented:
Thankyou @maobaolong ! Implemented the package __init__.py approach to resolve the error - 
Added __init__.py to benchmarks/, benchmarks/ttft-estimator/, lmcache/benchmarks/, and lmcache/benchmarks/ttft-estimator/ to make the source and packaged benchmarks proper packages.

Let me know if you prefer another approach or have any feedback!

@sammshen @maobaolong - sure, let me know when I can pick it up again

Navya1707 commented:
@maobaolong  - looks like renaming the script to ttft_estimator.py in both locations and updating the example usage strings accordingly will resolve this error. Please let me know if that looks good to you, I'll go ahead with the change

maobaolong commented:
@Navya1707 I'm ok for this fixing, but not sure we actually want this PR, refer to https://github.com/LMCache/LMCache/pull/2100#issuecomment-3584669685

Navya1707 commented:
Sure @maobaolong , thank you for looking into this! Please let me know if we're good to pick it up later

maobaolong commented:
@Navya1707 see https://github.com/LMCache/LMCache/issues/2066#issuecomment-3594178397

sammshen commented:
Thanks for the update @Navya1707 

Two things: 
1. please run the pre-commit. You should ignore the new `lmcache/benchmarks` folder in `pyproject.toml` for the relevant linteres
2. how is divergence between the two `/benchmarks` folder going to managed in the future?
